### DISCUSSION TITLE
"ScrollIntoView" method could be imporved?

### DISCUSSION THREAD

**Discussion by GHRyunosuke** [GCuser99/SeleniumVBA#134] [Ideas]
@GCuser99 and @6DiegoDiego9 

I found that "ScrollIntoView" is not working properly, what about just rewrite the method using ExecuteScript or similar?

`Call driver.ExecuteScript("arguments[0].scrollIntoView(true);", element) --> worked very well`
`element.ScrollIntoView --> does not work (it does scroll a tiny little bit...but way from scrolling the element into view)`

**Comment by GCuser99** [comment]
Thanks for reporting. I've fixed it and have a new commit ready soon with that and some other things.

**Reply by GHRyunosuke** [reply]
Great & thank you.

**Comment by GCuser99** [comment]
With v5.9, this issue has been solved. As discussed elsewhere, the scroll animation when smooth scrolling is async, meaning that subsequent commands can interfere with the scroll if it has not ended. The solution I've implemented is to wait until the `scollend` event is fired before moving on to next command via Javascript.
